OSVR-Unity/Assets/OSVRUnity/src/VRViewer.cs

```diff
@@ -0,0 +1,147 @@
+/// OSVR-Unity Connection
+///
+/// http://sensics.com/osvr
+///
+/// <copyright>
+/// Copyright 2014 Sensics, Inc.
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+/// </copyright>
+/// <summary>
+/// Author: Bob Berkebile
+/// Email: bob@bullyentertainment.com || bobb@pixelplacement.com
+/// </summary>
+
+using UnityEngine;
+using System.Collections;
+
+namespace OSVR
+{
+    namespace Unity
+    {
+        [RequireComponent(typeof(Camera))]
+        public class VRViewer : MonoBehaviour
+        {
+            #region Public Variables
+            public Camera Camera 
+            { 
+                get 
+                { 
+                    if(_camera == null)
+                    {
+                        _camera = GetComponent<Camera>();
+                    }
+                    return _camera; 
+                } 
+                set { _camera = value; } 
+            }
+            public DisplayController DisplayController { get { return _displayController; } set { _displayController = value; } }
+            [HideInInspector]
+            public Transform cachedTransform;
+            #endregion
+
+            #region Private Variables
+            private DisplayController _displayController;
+            private Camera _camera;
+            private bool disabledCamera = true;
+            #endregion
+
+            void Awake()
+            {
+                Init();
+            }
+
+            void Init()
+            {
+                //cache:
+                cachedTransform = transform;
+            }
+
+            void OnEnable()
+            {
+                StartCoroutine("EndOfFrame");
+            }
+
+            void OnDisable()
+            {
+                StopCoroutine("EndOfFrame");
+            }
+
+            //Updates the position and rotation of the head
+            public void UpdateViewerHeadPose(OSVR.ClientKit.Pose3 headPose)
+            {              
+                transform.localPosition = Math.ConvertPosition(headPose.translation);
+                transform.localRotation = Math.ConvertOrientation(headPose.rotation);
+            }
+
+            //Culling determines which objects are visible to the camera. OnPreCull is called just before this process.
+            void OnPreCull()
```

Comment:
This is the guts of the new implementation. Would like a review here. Do we also need to do anything in VRSurface::OnPreRender(), or will this suffice?